--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -8287,6 +8287,16 @@ async function BrowserOpenConnectionSettings() {
   b.click();
 }
 
+function BrowserJavaScriptSwitch(js_state) {
+  if (js_state) {
+    Services.prefs.setBoolPref("javascript.enabled", false);
+    window.alert("JavaScript disabled");
+  } else {
+    Services.prefs.setBoolPref("javascript.enabled", true);
+    window.alert("JavaScript enabled");
+  }
+}
+
 function AddKeywordForSearchField() {
   if (!gContextMenu) {
     throw new Error("Context menu doesn't seem to be open.");
--- a/browser/components/customizableui/CustomizableWidgets.jsm
+++ b/browser/components/customizableui/CustomizableWidgets.jsm
@@ -587,6 +587,39 @@ let connectionSettingsButton = {
 };
 CustomizableWidgets.push(connectionSettingsButton);
 
+const jsIcon1 = "chrome://global/skin/icons/security-warning.svg";
+const jsIcon2 = "chrome://global/skin/icons/security.svg";
+let javaScriptButton = {
+  id: "javascript-button",
+  l10nId: "toolbar-javascript-button",
+  onCreated(aNode) {
+    // Setup icon on startup
+    let js_state = Services.prefs.getBoolPref("javascript.enabled");
+
+    if (js_state) {
+      aNode.setAttribute("image", jsIcon1);
+    } else {
+      aNode.setAttribute("image", jsIcon2);
+    }
+  },
+  onCommand(aEvent) {
+    let win = aEvent.target.ownerGlobal;
+    let js_state = Services.prefs.getBoolPref("javascript.enabled");
+    win.BrowserJavaScriptSwitch(js_state);
+
+    // Setup icon after switch
+    let js_state_new = Services.prefs.getBoolPref("javascript.enabled");
+    let widget = CustomizableUI.getWidget("javascript-button").instances;
+
+    if (js_state_new) {
+      widget[0].node.setAttribute("image", jsIcon1);
+    } else {
+      widget[0].node.setAttribute("image", jsIcon2);
+    }
+  },
+};
+CustomizableWidgets.push(javaScriptButton);
+
 if (Services.prefs.getBoolPref("privacy.panicButton.enabled")) {
   CustomizableWidgets.push({
     id: "panic-button",
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -685,6 +685,10 @@ toolbar-connection-settings-button =
   .label = Connection Settings
   .tooltiptext = Open connection settings
 
+toolbar-javascript-button =
+  .label = JavaScript Switch
+  .tooltiptext = Disable/Enable JavaScript
+
 toolbar-overflow-customize-button =
   .label = Customize toolbar…
   .accesskey = C
--- a/browser/themes/shared/toolbarbutton-icons.inc.css
+++ b/browser/themes/shared/toolbarbutton-icons.inc.css
@@ -314,6 +314,10 @@ toolbar {
   list-style-image: url("chrome://global/skin/icons/settings.svg");
 }
 
+#javascript-button {
+  list-style-image: url("chrome://global/skin/icons/security-warning.svg");
+}
+
 #PanelUI-menu-button {
   list-style-image: url("chrome://browser/skin/menu.svg");
 }
